gcc/rust/hir/rust-ast-lower-type.h

```diff
@@ -79,7 +80,14 @@ class ASTLoweringType : public ASTLoweringBase
   void visit (AST::TraitObjectType &type) override;
 
 private:
-  ASTLoweringType () : ASTLoweringBase (), translated (nullptr) {}
+  ASTLoweringType (bool default_to_static_lifetime)
+    : ASTLoweringBase (),
+      default_to_static_lifetime (default_to_static_lifetime),
+      translated (nullptr)
+  {}
+
+  /** Used when compiling const and static items. */
+  bool default_to_static_lifetime;
```

Comment:
I don't get how this flag is used or set. If we are lowering an `AST::ConstantItem` or `AST::StaticItem`, we know we are doing so and can pass an explicit `true` flag to `ASTLoweringType` (or even better, a `LifetimeDefault::Static` enum variant which is clearer). and we could have a default value to the bool argument to the `ASTLoweringType` argument?

does that make sense? or is it important that we keep it this way for future pull-requests?

Reply:
This flag is used to get the state across the visitor API and propagate further `ASTLoweringType::translate` calls since the visitor API does not allow more parameters. I think it makes sense. Does it make sense now?

Reply:
ah, yeah I see what you mean. so we really can't do without a default parameter then?

Reply:
Do you mean the member variable or the parameter? We need the member variable, the parameter is useful because we don't need to change it everywhere.

Reply:
I meant the parameter - but I'm fine with keeping it this way, your explanation makes sense. I'll merge this now